app/forms/waste_exemptions_engine/contact_address_manual_form.rb

```diff
@@ -2,62 +2,37 @@
 
 module WasteExemptionsEngine
   class ContactAddressManualForm < BaseForm
-    attr_accessor :address_finder_error
-    attr_accessor :contact_address
-    attr_accessor :premises, :street_address, :locality, :postcode, :city
+    delegate :premises, :street_address, :locality, :city, to: :contact_address, allow_nil: true
+    delegate :contact_address, :business_type, to: :transient_registration
+
+    attr_accessor :postcode, :address_finder_error
 
     validates :contact_address, "waste_exemptions_engine/manual_address": true
 
     after_initialize :setup_postcode
 
     def submit(params)
-      permitted_attributes = params.require(:contact_address)
-      permitted_attributes = permitted_attributes.permit(:locality, :postcode, :city, :premises, :street_address, :mode)
-
-      # Needed for validation
-      assign_params(permitted_attributes)
-
-      super(contact_address_attributes: permitted_attributes)
+      super(contact_address_attributes: params[:contact_address])
     end
 
     private
 
     def setup_postcode
       self.postcode = transient_registration.temp_contact_postcode
 
-      self.contact_address = transient_registration.contact_address
-
       # Check if the user reached this page through an Address finder error.
       # Then wipe the temp attribute as we only need it for routing
       self.address_finder_error = transient_registration.address_finder_error
       transient_registration.update_attributes(address_finder_error: nil)
 
       # Prefill the existing address unless the postcode has changed from the existing address's postcode
-      prefill_contact_address_params if saved_address_still_valid?
-    end
-
-    def assign_params(permitted_attributes)
-      self.contact_address = transient_registration.build_contact_address(permitted_attributes)
-
-      prefill_contact_address_params
+      transient_registration.contact_address = nil unless saved_address_still_valid?
     end
 
     def saved_address_still_valid?
       return false unless contact_address
-      return true if postcode.blank?
-      return true if postcode == contact_address.postcode
-
-      false
-    end
-
-    def prefill_contact_address_params
-      return unless contact_address
 
-      self.premises = contact_address.premises&.strip
-      self.street_address = contact_address.street_address&.strip
-      self.locality = contact_address.locality&.strip
-      self.city = contact_address.city&.strip
-      self.postcode = contact_address.postcode&.strip
+      postcode == contact_address.postcode
```

Comment:
I finally figured that here instead of using `attr_accessor` for the address we can still do delegate as per the rest of the forms. The `attr_accessor` was still there as we have to keep an address stored BUT not replay it in case the user have gone back and changed the postcode. In order to achieve that same behaviour without the `attr_accessor` assignment of params, I have only kept the check for `address_still_valid` and set the address to `nil` in the in-memory transient_registration when that check is `false`. Same behaviour, but the code is in-line with the rest we have now.